### PR TITLE
feat(heroes): enforce minimum hero name length of 31 characters

### PR DESCRIPTION
**Pull request by danielmackay**
## Summary

- Adds `Hero.NameMinLength = 31` domain invariant with `ThrowIfLessThan` guard in the `Name` setter
- Wires `MinimumLength(Hero.NameMinLength)` into `CreateHeroRequestValidator`
- Updates all test fixtures, factories, and Swagger examples to use names ≥ 31 chars

## Changes

Issue #226 required that hero names be longer than 30 characters. The rule is enforced at two layers: the request validator rejects bad input early with a 400, and the domain entity guards the invariant so any future caller that bypasses the endpoint is also protected. Existing tests using short names like "Clark Kent" and "hero1" were updated to comply.

## Test Plan

- [ ] `dotnet test tests/WebApi.UnitTests` — 38 tests pass, including 4 new boundary cases (30 throws, 31 succeeds, 100 succeeds, 101 throws)
- [ ] `dotnet test tests/WebApi.IntegrationTests` — 11 tests pass, including new `Command_WithNameTooShort_ShouldReturn400`
- [ ] POST to `/heroes` with a name ≤ 30 chars → expect HTTP 400 with validation error on `Name`
- [ ] POST to `/heroes` with a name ≥ 31 chars → expect HTTP 200

### CODE REVIEW

**Review by danielmackay — COMMENTED** [review]
Test review

File: src/WebApi/Features/Heroes/Common/MyHeroesService.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+namespace SSW.VerticalSliceArchitecture.Features.Heroes;
+
+public class MyHeroesService
+{
```
Comment (danielmackay) [inline src/WebApi/Features/Heroes/Common/MyHeroesService.cs:4]: Test comment

**Review by danielmackay — COMMENTED** [review]
Test review 2

File: src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs [hunk 2]
```diff
@@ -0,0 +1,6 @@
+namespace SSW.VerticalSliceArchitecture.Features.Heroes.Endpoints;
+
+public class CreateHeroCommand
+{
```
Comment (danielmackay) [inline src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs:4]: Test comment

File: src/WebApi/Features/Heroes/Common/MyHeroesService.cs (hunk 1 above)
Comment (danielmackay) [inline src/WebApi/Features/Heroes/Common/MyHeroesService.cs:4]: [Critical] Empty stub accidentally included in this PR — the class has no implementation and no relation to the min-length feature. Remove this file or implement it in a separate PR.

File: src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs (hunk 2 above)
Comment (danielmackay) [inline src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs:4]: [Critical] Empty stub accidentally included in this PR. The namespace also says `Endpoints` but the file lives under `CreateHero/`. Remove this file or implement it in a separate PR.

**Review by danielmackay — COMMENTED** [review]
Two blocking issues need resolving before merge. The core invariant implementation (dual-layer enforcement via domain guard + FluentValidation) is correct and the boundary test coverage is solid — just clean up the accidental commits and finish the restructure.

---

**[Critical] Incomplete file restructuring — `UpdateHeroEndpoint.cs`**
`CreateHeroEndpoint` and `GetAllHeroesEndpoint` were moved into feature-specific subfolders (`CreateHero/`, `GetAllHeroes/`), but `UpdateHeroEndpoint` was left in the shared `Endpoints/` folder. Either move all endpoints to their own feature folders or revert the moves — a half-finished restructure makes the codebase harder to navigate than either consistent state.

File: src/WebApi/Features/Heroes/Common/MyHeroesService.cs (hunk 1 above)
Comment (danielmackay) [inline src/WebApi/Features/Heroes/Common/MyHeroesService.cs:4]: [Critical] Empty stub accidentally included in this PR — the class has no implementation and no relation to the min-length feature. Remove this file or implement it in a separate PR.

File: src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs (hunk 2 above)
Comment (danielmackay) [inline src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs:4]: [Critical] Empty stub accidentally included in this PR. The namespace also says `Endpoints` but the file lives under `CreateHero/`. Remove this file or implement it in a separate PR.

**Review by danielmackay — COMMENTED** [review]
Two blocking issues need resolving before merge. The core invariant implementation (dual-layer enforcement via domain guard + FluentValidation) is correct and the boundary test coverage is solid — just clean up the accidental commits and finish the restructure.

---

**[Critical] Incomplete file restructuring — `UpdateHeroEndpoint.cs`**
`CreateHeroEndpoint` and `GetAllHeroesEndpoint` were moved into feature-specific subfolders (`CreateHero/`, `GetAllHeroes/`), but `UpdateHeroEndpoint` was left in the shared `Endpoints/` folder. Either move all endpoints to their own feature folders or revert the moves — a half-finished restructure makes the codebase harder to navigate than either consistent state.

File: src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs (hunk 2 above)
Comment (danielmackay) [reply src/WebApi/Features/Heroes/CreateHero/CreateHeroCommand.cs:4]: yeah, this is right, but do it like this instead...

**Comment by danielmackay** [comment]
Used in a brown bag session. Not a real PR.
